src/Network/HTTP/Client/Overrides.hs

```diff
@@ -0,0 +1,90 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Network.HTTP.Client.Overrides (
+    -- ** HTTP client overrides
+      withHttpClientOverrides
+    , withHttpClientOverridesFile
+    , httpClientOverrides
+    , parseConfigFile
+    -- ** Types
+    , ConfigFile(..)
+    , Config(..)
+    , LogOptions(..)
+    , LogFormat(..)
+    , RequestOverride(..)
+    , URL(..)
+    ) where
+
+import Data.Either (either)
+import Data.List (find)
+import GHC.Stack (HasCallStack)
+import System.Environment (lookupEnv)
+
+import Data.Yaml (ParseException, decodeFileEither, prettyPrintParseException)
+import Network.HTTP.Client (ManagerSettings, Response, Request)
+import qualified Network.HTTP.Client as HTTP
+
+import Network.HTTP.Client.Overrides.Internal.Logger
+import Network.HTTP.Client.Overrides.Internal.RequestOverrides
+import Network.HTTP.Client.Overrides.Internal.Types
+
+------------------------
+-- HTTP client overrides
+------------------------
+
+-- | If the @HTTP_CLIENT_OVERRIDES@ environment variable is set, this function
+-- reads the specified file as a 'ConfigFile' and applies the overrides to
+-- 'ManagerSettings'.
+withHttpClientOverrides :: HasCallStack => ManagerSettings -> IO ManagerSettings
+withHttpClientOverrides manager = do
+    filepath <- lookupEnv "HTTP_CLIENT_OVERRIDES"
```

Comment:
Being able to set the path programmatically would be neat as well I think.

So you'd have three API functions in total

- `withHttpClientOverrides`: Reads the `HTTP_CLIENT_OVERRIDES` var.
- `withHttpClientOverridesFile`: Uses a `ConfigFile` as an argument.
- `withHttpClientOverridesFilePath`: Would accept a `FilePath` argument.

Reply:
That's the idea behind combining `parseConfigFile` and `withHttpClientOverridesFile` which also allows the user to handle errors properly when parsing the config file. The function that uses an environment variable throws an exception if the config file fails to parse (I should make this clearer in the docs).

Reply:
I've improved the docs and added `withHttpClientOverridesThrow` to make it clear that the function might throw an exception.